stack/app/agents/crag_action_executor_v1.py

```diff
@@ -0,0 +1,245 @@
+from typing import List, TypedDict, Sequence, Optional, Union, Dict, Any
```

Comment:
This is not currently in use - it is here because it contains a potential solution for streaming the crag intermediate steps.